include/eve/arch/cpu/wide.hpp

```diff
@@ -401,9 +401,9 @@ namespace eve
     //! @brief Performs a compound bitwise and on all the wide lanes and assign the result to the current
     //! one
     template<value V>
-    friend EVE_FORCEINLINE auto operator&=(wide& w, V o) noexcept
+    friend EVE_FORCEINLINE bit_value_t<wide, V>& operator&=(wide& w, V o) noexcept
```

Comment:
I have a problem with this style here a bit.

You see - it was confusing for me, what is `bit_value_t`, so what can you expect from a casual reader of this file (or the error message).

This is `wide`, right? You can only do bit operations on the same type `wide` and scalar of the same type?

I believe the result type then should be a wide and requires clause explicitly specify when the operator is enabled.

Reply:
I think it doesn't mean anything. You want a concept with a meaning.